lib/nolibc/README.md

```diff
@@ -0,0 +1,69 @@
+# nolibc, Unikraft's Lightweight Subset of the C Standard Library
+
+The `nolibc` library provides the basic core functionalities of libc in a lighter and more compact package, fulfilling the needs of most simple applications without occupying a large amount of memory, while also remaining POSIX-compliant.
+Complex applications that require a more comprehensive set of standard C functions are better suited with Unikraft's port of [musl](https://github.com/unikraft/lib-musl).
+
+## Configuring Applications to Use `nolibc`
+
+The `nolibc` library is an internal library of Unikraft and is enabled by default when configuring a new project.
+Opting for a different implementation of the C standard library will automatically override `nolibc` and utilize the one provided by the user.
+This behavior can be observed in the configuration menu under the `Library Configuration` section.
+
+An example of a simple application that uses the `nolibc` library is the following:
+
+```c
+#include <stdio.h>
+#include <string.h>
+
+int main(int argc, char *argv[])
+{
+    char str[1024], *p;
+
+    strcpy(str, "There's the way it ought to be");
+    printf("%s\n", str);
+
+    strcat(str, "and there's the way it is");
+    p = strchr(str, 'b');
+    p += 2;
+    fprintf(stderr, "%s\n", p);
+
+    return 0;
+}
+```
+
+The steps to configure the application to use `nolibc` are as follows:
+
+1. Enter the configuration interface with:
+
+    ```console
+    $ make menuconfig
+    ```
+
+1. Enter the `Library Configuration` submenu
+1. Select `nolibc:` `Only necessary subset of libc functionality`
+1. Choose between using Unikraft-specific assertions or libc assertions by toggling `Implement assertions with libukdebug`
```

Comment:
Rephrase this so it does not sound like something the user should do, but as something that the user could do (i.e. something like `You can switch between [Unikraft-specific assertions](link to ukdebug assertion header file) or [nolibc assertions](link to the nolibc assert.h header file) by toggling the Implement assertions with libukdebug option`).